plugins/sqlutil/sqlutils.go

```diff
@@ -62,3 +65,23 @@ func BuildBoolMap(strList []string) map[string]bool {
 
 	return boolMap
 }
+
+func OpenWithOtel(driverName, connectionURL string, otelSemConv attribute.KeyValue) (db *sql.DB, err error) {
+	driverName, err = otelsql.Register(driverName,
+		otelsql.TraceQueryWithoutArgs(),
+		otelsql.TraceRowsClose(),
+		otelsql.TraceRowsAffected(),
+		otelsql.WithSystem(otelSemConv),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("register %s otelsql wrapper: %w", driverName, err)
+	}
+
+	db, err = sql.Open(driverName, connectionURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = otelsql.RecordStats(db)
+	return db, err
```

Comment:
We should always return either a valid result with nil err or zero value result with err. 
```suggestion
	if err := otelsql.RecordStats(db); err != nil {
		return nil, err
	}

	return db, nil
```